### PR TITLE
Setting up loading button

### PR DESCRIPTION
**Pull request by jumaallan**
Story: N/

## Summary

Adds a loading button which shows text or a circular loading indicator depending on the state of the app

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

Screenshot 2023-07-20 at 12 07 22 PM

### CODE REVIEW

File: lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.unit.dp
+import com.smileidentity.R
+import com.smileidentity.compose.preview.SmilePreviews
+
+@Composable
+fun LoadingButton(
+    buttonText: String,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
```
Comment (jumaallan) [inline lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt:21]: I have removed this in the next PR. If loading, we disable button so no need to pass two params

File: lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt:1]: Can you rename the file to match the component name? `LoadingButton.kt`

File: lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                        .align(Alignment.Center),
+                )
+            }
+        }
+    }
+}
+
+@SmilePreviews
+@Composable
+fun LoadingIndicatorButtonPreview() {
```
Comment (vanshg) [inline lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt:49]: Same here
```suggestion
fun LoadingButtonPreview() {
```

File: lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt (hunk 1 above)
Comment (vanshg) [reply lib/src/main/java/com/smileidentity/compose/components/LoadingIndicatorButton.kt:21]: Agreed

**Review by vanshg — CHANGES_REQUESTED** [review]
Can you add a test for this since it's in the SDK?

**Review by vanshg — APPROVED** [review]
